src/components/AddDevice/AddDevice.tsx

```diff
@@ -180,6 +180,7 @@ function SignInPage() {
       const public_key =  decodeIfTruthy(searchParams.get('public_key'));
       const contract_id = decodeIfTruthy(searchParams.get('contract_id'));
       const methodNames = decodeIfTruthy(searchParams.get('methodNames'));
+      const gateway = decodeIfTruthy(searchParams.get('gateway'));
```

Comment:
`gateway` should be inferred from `success_url`. IMO we can use the domain name here

Reply:
Actually good idea!, will make update shortly